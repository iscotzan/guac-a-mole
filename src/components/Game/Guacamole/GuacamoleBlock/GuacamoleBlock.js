import React from "react";

const GuacamoleBowl = props => {
  return (
    <div className="game-block" style={{ display: props.context.display }}>
      <div className="game__whack">
        <button
          className={props.currentClass}
          onClick={props.onClick.bind(
            this,
            props.bowlNumber,
            props.context.role
          )}
          disabled={props.disabled}
          style={{
            WebkitTransform: props.context.style
          }}
        />
      </div>

      <div className="game-plate" />
    </div>
  );
};

export default GuacamoleBowl;
