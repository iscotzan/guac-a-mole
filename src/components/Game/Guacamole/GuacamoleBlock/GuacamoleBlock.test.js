import React from "react";
import GuacamoleBlock from "./GuacamoleBlock";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const pawns = {
  0: {
    role: "avocado",
    active: false,
    targetHit: false,
    lifetime: 0,
    style: "translate(0, -110%)"
  }
};

const wrapper = shallow(
  <GuacamoleBlock
    context={pawns[0]}
    onClick={() => {}}
    bowlNumber={0}
    currentClass={"avocado"}
    disabled={pawns[0].targetHit}
  />
);
describe("<GuacamoleBlock />", () => {
  it("should be defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("renders without crashing", () => {
      expect(toJson(wrapper)).toMatchSnapshot();
  });
});
