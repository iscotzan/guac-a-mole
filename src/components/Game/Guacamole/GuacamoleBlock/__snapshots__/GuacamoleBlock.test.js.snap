// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GuacamoleBlock /> renders without crashing 1`] = `
<div
  className="game-block"
  style={
    Object {
      "display": undefined,
    }
  }
>
  <div
    className="game__whack"
  >
    <button
      className="avocado"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "WebkitTransform": "translate(0, -110%)",
        }
      }
    />
  </div>
  <div
    className="game-plate"
  />
</div>
`;
